tests/integration-service/integration-with-env.go

```diff
@@ -0,0 +1,171 @@
+					Expect(err).ShouldNot(HaveOccurred())
+					if controllerutil.ContainsFinalizer(pipelineRun, pipelinerunFinalizerByIntegrationService) {
+						return fmt.Errorf("build pipelineRun %s/%s still contains the finalizer: %s", pipelineRun.GetNamespace(), pipelineRun.GetName(), pipelinerunFinalizerByIntegrationService)
+					}
+					return nil
+				}, timeout, interval).Should(Succeed(), "timeout when waiting for finalizer to be removed")
+			})
+
+			It(fmt.Sprintf("checks if CronJob %s exists", spaceRequestCronJobName), func() {
+				spaceRequestCleanerCronJob, err := f.AsKubeAdmin.CommonController.GetCronJob(spaceRequestCronJobNamespace, spaceRequestCronJobName)
+				Expect(err).ShouldNot(HaveOccurred())
+				Expect(strings.Contains(spaceRequestCleanerCronJob.Name, spaceRequestCronJobName)).Should(BeTrue())
+			})
+
+			It("checks if all of the integrationPipelineRuns passed", Label("slow"), func() {
+				Expect(f.AsKubeDeveloper.IntegrationController.WaitForAllIntegrationPipelinesToBeFinished(testNamespace, applicationName, snapshot)).To(Succeed())
+			})
+
+			It("checks if space request is created in namespace", func() {
+				spaceRequestsList, err := f.AsKubeAdmin.GitOpsController.GetSpaceRequests(testNamespace)
+				Expect(err).ShouldNot(HaveOccurred())
+
+				Expect(spaceRequestsList.Items).ToNot(BeEmpty(), "Expected spaceRequestsList.Items to have at least one item")
+				spaceRequest = &spaceRequestsList.Items[0]
+				Expect(strings.Contains(spaceRequest.Name, spaceRequestNamePrefix)).Should(BeTrue())
+			})
+
+			It("checks if the passed status of integration test is reported in the Snapshot", func() {
+				timeout = time.Second * 240
+				interval = time.Second * 5
+				Eventually(func() error {
+					snapshot, err = f.AsKubeAdmin.IntegrationController.GetSnapshot(snapshot.Name, "", "", testNamespace)
+					Expect(err).ShouldNot(HaveOccurred())
+
+					statusDetail, err := f.AsKubeDeveloper.IntegrationController.GetIntegrationTestStatusDetailFromSnapshot(snapshot, integrationTestScenario.Name)
+					Expect(err).ToNot(HaveOccurred())
+
+					if statusDetail.Status != intgteststat.IntegrationTestStatusTestPassed {
+						return fmt.Errorf("test status for scenario: %s, doesn't have expected value %s, within the snapshot: %s", integrationTestScenario.Name, intgteststat.IntegrationTestStatusTestPassed, snapshot.Name)
+					}
+					return nil
+				}, timeout, interval).Should(Succeed())
+			})
+
+			It("checks if the finalizer was removed from all of the related Integration pipelineRuns", func() {
+				Expect(f.AsKubeDeveloper.IntegrationController.WaitForFinalizerToGetRemovedFromAllIntegrationPipelineRuns(testNamespace, applicationName, snapshot)).To(Succeed())
+			})
+
+			It("checks that when deleting integration test scenario pipelineRun, spaceRequest is deleted too", func() {
+				integrationPipelineRun, err = f.AsKubeAdmin.IntegrationController.GetIntegrationPipelineRun(integrationTestScenario.Name, snapshot.Name, testNamespace)
```

Comment:
I have the same concern as build PLR if the plr is deleted quickly, but I don't know how soon the PLR can be deleted once finalizer is removed.

Reply:
then the test will fail and we have to investigate to find root cause